pkg/webhook/conversion/testdata/main.go

```diff
@@ -46,10 +47,19 @@ func init() {
 
 func main() {
 	var metricsAddr string
-	var enableLeaderElection bool
+	var enableLeaderElection *bool
 	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
-	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
-		"Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.")
+	flag.Func("enable-leader-election",
+		"Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.",
+		func(flagVal string) error {
+			boolVal, err := strconv.ParseBool(flagVal)
+			if err != nil {
+				return err
+			}
+			enableLeaderElection = &boolVal
+			return nil
+		},
```

Comment:
I wonder if we could provide a helper for this since this is a big chunk people will have to repeat.

Reply:
I think the main code lines the users will repeat is to convert normal bool var to pointer which can be done by k8s util `pointer.BoolPtr()`. If we think about it in a generic way this function is taking the string and assigning/returning the bool ptr and error `IF` its a valid string. If user wants that they can directly `strong.ParseBool()`

Or I am I missing some point?

Reply:
Why do we need to change the flag?

Reply:
We can probably just use the plain bool, and dereference it, or use pointer.BoolPtr(enableLeaderElection) instead.

Reply:
That was the whole point for the change actually. If we keep the flag as plain bool, we won't be able to differentiate if the user explicitly gave it false or it was the zero value of that variable.

If user gave it false then we have to pass it down to the `Options` so that the config file option is not considered. If user did not give anything for this flag, then we will consider the config file option

Reply:
I was more thinking of a helper function so every controller that currently uses:

```golang
flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
		"Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.")
```

is now going to have to repeat your flap parsing func, while not hard it's breaking and something folks will have to do when upgrading, thus it could be nice to have a helper func people could define here instead. Something like:

```golang
var enableLeaderElection *bool
flag.Func("enable-leader-election",
		"Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.",
		ctrl.BoolPointerFlag(enableLeaderElection),
)
```

then in `pkg/manager` we could have `BoolPointerFlag()` as a helper.

```golang
func BoolPointerFlag(value *bool) func(flagVal string) error {
	return func(flagVal string) error {
		boolVal, err := strconv.ParseBool(flagVal)
		if err != nil {
			return err
		}
		value = &boolVal
		return nil
	}
}
```

Might be unnecessary 🤷‍♂️

Reply:
Does returning an error from BoolPointerFlag actually do what you want this to do? See https://play.golang.org/p/-VBOv0ZkZ1k, if the string is empty, the bool conversion fails

Reply:
@vincepri Yeah I understood your point. So the current behaviour would be following:
- If `--enable-leader-election` is not passed -> the function block to pointer conversion won't run
- If `--enable-leader-election` is passed but with either empty string `""` or with some invalid string -> It will fail with showing the usage message
- If `--enable-leader-election` is passed with either `true/false` -> it will accept and move ahead

Reply:
@christopherhein I got your point. I will add that helper